src/asset.ts

```diff
@@ -21,14 +21,25 @@ export default class Asset {
     ethAddress: string,
     contractAddress?: string,
   ) {
-    const data = { appId, timestamp: Date.now() };
+    const timestamp = Date.now();
+    const query = { appId };
+    const trailingUrl = contractAddress
+      ? `nft/${chainId}/${ethAddress}/${contractAddress}`
+      : `nft/${chainId}/${ethAddress}`;
+    const data = {
+      method: 'GET',
+      path: `/asset/${trailingUrl}`,
+      timestamp,
+      querystring: stringify(query),
+    };
     const signature = this.ain.wallet.sign(stringify(data));
```

Comment:
would it be useful to add a util function like `signQueryData(data: any)` for readability even though it's a one-line function?

Reply:
Done! Add `signData` function in each class.